client/signup/steps/subscribing-email/index.jsx

```diff
@@ -1,53 +1,72 @@
-// import { getTracksAnonymousUserId } from '@automattic/calypso-analytics';
-// import config from '@automattic/calypso-config';
+import { getTracksAnonymousUserId } from '@automattic/calypso-analytics';
+import config from '@automattic/calypso-config';
+import page from '@automattic/calypso-router';
 import { useEffect, useState } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
 import emailValidator from 'email-validator';
 import { localize } from 'i18n-calypso';
 import { connect } from 'react-redux';
-// import { getLocaleSlug } from 'calypso/lib/i18n-utils';
+import { getLocaleSlug } from 'calypso/lib/i18n-utils';
+import { isExternal } from 'calypso/lib/url';
 import StepWrapper from 'calypso/signup/step-wrapper';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { submitSignupStep } from 'calypso/state/signup/progress/actions';
 import SubscribingEmailStepContent from './content';
 
-// const createNewAccount = async ( props, setIsLoading, setDisplayUserExistsError ) => {
-// 	const email = props.queryParams.email;
-// 	try {
-// 		// eslint-disable-next-line no-undef
-// 		await wpcom.req.post( '/users/new', {
-// 			email: typeof email === 'string' ? email.trim() : '',
-// 			is_passwordless: true,
-// 			signup_flow_name: props.flowName,
-// 			validate: false,
-// 			locale: getLocaleSlug(),
-// 			client_id: config( 'wpcom_signup_id' ),
-// 			client_secret: config( 'wpcom_signup_key' ),
-// 			anon_id: getTracksAnonymousUserId(),
-// 		} );
-// 		// Do stuff with response and redirect
-// 		props.recordTracksEvent( 'calypso_signup_reader_landing_cta' );
-// 		props.submitSignupStep( { stepName: props.stepName } );
-// 		props.goToNextStep();
-// 	} catch ( error ) {
-// 		if ( ! [ 'already_taken', 'already_active', 'email_exists' ].includes( error.error ) ) {
-// 			// Subscribe existing user to guides emails through API endpoint
-// 		}
-// 		setIsLoading( false );
-// 	}
-// };
+const createNewAccount = async ( {
```

Comment:
Heya @jeyip Have you considered using a React Query [mutation](https://tanstack.com/query/latest/docs/framework/react/guides/mutations) for this? It should be possible to create a simple reusable hook from the looks of it and avoid some of the boilerplate (awaits, try/catch clause, etc.)

Reply:
Hey thanks for checking in @chriskmnds. I was actually planning on updating and using an abstraction I found here https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/signup/api/account.tsx#L61-L89, but avoiding the try/catch and await is a good point.

I'll look into `react-query` 👍

Reply:
[4cd92e4](https://github.com/Automattic/wp-calypso/pull/92013/commits/4cd92e447f81f54f257225e43d9379cf1dfa4197)

Reply:
Nice. Either way sounded great. Perhaps the new abstraction/hook can propagate to become the main form.